src/ResourceManager/RecoveryServices.Backup/Cmdlets/Jobs/GetAzureRmRecoveryServicesJob.cs

```diff
@@ -0,0 +1,145 @@
+        [ValidateNotNullOrEmpty]
+        public string Operation { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = ParamHelpMsg.Job.StatusFilter)]
+        [ValidateNotNullOrEmpty]
+        public string Status { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            ExecutionBlock(() =>
+            {
+                base.ExecuteCmdlet();
+
+                // initialize values to default
+                DateTime rangeStart = DateTime.UtcNow.AddDays(-1);
+                DateTime rangeEnd = DateTime.UtcNow;
+
+                if (From.HasValue)
+                {
+                    rangeStart = From.Value;
+                }
+
+                if (To.HasValue)
+                {
+                    rangeEnd = To.Value;
+                }
+
+                // validate filters
+                if (rangeEnd >= rangeStart)
+                {
+                    throw new Exception(CmdletWarningAndErrorMessages.Job.ToShouldBeLessThanFrom);
+                }
+                else if (rangeEnd.Subtract(rangeStart) > TimeSpan.FromDays(30))
+                {
+                    throw new Exception(CmdletWarningAndErrorMessages.Job.AllowedDateTimeRangeExceeded);
+                }
+
+                // validate JobId and Job objects
+                if (!string.IsNullOrEmpty(JobId))
+                {
+                    // if JobId and Job are provided together and they don't match then throw exception
+                    if (Job != null && JobId != Job.InstanceId)
+                    {
+                        throw new Exception(CmdletWarningAndErrorMessages.Job.JobIdAndJobMismatch);
+                    }
+                }
+                else if (Job != null)
+                {
+                    JobId = Job.InstanceId;
+                }
+
+                // TODO: Initialize Vault object from base cmdlet once support is added
+                List<AzureRmRecoveryServicesJobBase> result = new List<AzureRmRecoveryServicesJobBase>();
+                int resultCount = 0;
+                var adapterResponse = HydraAdapter.GetJobs(Vault.ResouceGroupName, Vault.Name, JobId, Status, Operation, rangeStart, rangeEnd,
+                    BackupManagementType);
+                JobConversions.AddHydraJobsToPSList(adapterResponse, result, ref resultCount);
+
+                while (!string.IsNullOrEmpty(adapterResponse.ItemList.NextLink))
+                {
+                    if (resultCount >= JobConstants.MaximumJobsToFetch)
+                    {
+                        // trace a warning that there are more jobs and user has to refine filters.
+                        WriteWarning(CmdletWarningAndErrorMessages.Job.RefineFilters);
+                        break;
+                    }
+
+                    string skipToken;
+                    HydraHelpers.GetSkipTokenFromNextLink(adapterResponse.ItemList.NextLink, out skipToken);
```

Comment:
this is generic logic for all feature - iterate until nextLink is empty. Better to have one generic templatized method in helper which takes delegate/action as input for every feature